scripts/modal.js

```diff
@@ -0,0 +1,56 @@
+export function openModal(book, section) {
+  const mainModal = document.createElement("div");
+  mainModal.className = "modal";
+
+  //const nav = document.createElement("nav");
+  const modalClose = document.createElement("button");
+  modalClose.textContent = "X";
+
+  //mainModal.appendChild(nav);
+
+  modalClose.addEventListener("click", () => {
+    mainModal.style.display = "none";
+  });
+
+  const modalBookInfoContainer = document.createElement("div");
+  const modalBookTitle = document.createElement("h4");
+  const modalSubtitle = document.createElement("h4");
+  const modalImage = document.createElement("img");
+  const modalAuthor = document.createElement("h4");
+  const modalDes = document.createElement("p");
+
+  //Classes assigned to HTML tags
+  modalBookInfoContainer.className = "modal__container";
+  modalClose.className = "modal__container__button";
+  modalBookTitle.className = "modal__container__title";
+  modalSubtitle.className = "modal__container__subtitle";
+  modalImage.className = "modal__container__img";
+  modalDes.className = "modal__container__desc";
+
+  //Adding content to modal
+  modalBookInfoContainer.appendChild(modalClose);
+  modalBookTitle.textContent = book.volumeInfo.title;
+  modalSubtitle.textContent = book.volumeInfo.subtitle;
+  modalImage.src = book.volumeInfo.imageLinks.thumbnail;
+  const modalAuthorName = book.volumeInfo.authors;
+  for (const author of modalAuthorName) {
+    const listItem = document.createElement("h4");
+    listItem.textContent = author;
+    modalAuthor.appendChild(listItem);
+  }
+  modalDes.textContent = book.volumeInfo.description;
+
+  //Add HTML Tags to its parent din
+  modalBookInfoContainer.appendChild(modalImage);
+  modalBookInfoContainer.appendChild(modalBookTitle);
+  modalBookInfoContainer.appendChild(modalSubtitle);
+  modalBookInfoContainer.appendChild(modalAuthor);
+  modalBookInfoContainer.appendChild(modalDes);
+
+  //append to mainmodal
+  mainModal.appendChild(modalBookInfoContainer);
+
+  mainModal.style.display = "flex";
+
+  section.appendChild(mainModal);
+}
```

Comment:
try and reduce the code by having a function that creayes elements, here's some pseudo code of what that could look like: `
const createBookCard = (elArr, classArr, parent) => {

    // create all elements you need for a book card
    const elements = elArr.forEach((el) => document.createElement(el));

    // add class names to them
    elements.forEach((el, i) => (el.className = classArr[i]));

    // tell them where to go on a page
    elements.forEach((el) => parent.appendChild(el));
};`